### PR TITLE
RBAC: Add option to skip rbac check for specified verbs

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
Add `Unchecked` options to `ResourceAuthorizerOptions`. This option takes precedence over `Mapping` and can be used to skip authorization check for specified verbs. This is then used for `Display` resource for get and list allowing all authenticated users to resolve display result for provided keys.

A lot of this pr is renaming to make it more consistent with how other resources are structured but the relevant changes are done in pkg/services/accesscontrol/authorizer.go and pkg/registry/apis/iam/authorizer.go.

I also removed feature toggle check for experimental apis. We now have versioned apis and these are all v0alpha1

**Which issue(s) does this PR fix?**:
Part of: https://github.com/grafana/identity-access-team/issues/889

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/apis/iam/v0alpha1/types_display.go [hunk 1]
```diff
@@ -7,23 +7,24 @@ import (
...
-	// +listType=atomic
-	Display []IdentityDisplay `json:"display"`
-
 	// Input keys that were not useable
 	// +listType=set
 	InvalidKeys []string `json:"invalidKeys,omitempty"`
+
+	// Matching items (the caller may need to remap from keys to results)
+	// +listType=atomic
+	Items []Display `json:"display"`
```
Comment (kalleep) [inline pkg/apis/iam/v0alpha1/types_display.go:24]: Renamed this to `Items` to make it more consistent with other list responses

**Review by gamab — APPROVED** [review]
LGTM!

File: pkg/registry/apis/iam/user/rest_display.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ func NewLegacyDisplayREST(store legacy.LegacyIdentityStore) *LegacyDisplayREST {
 }
 
 func (r *LegacyDisplayREST) New() runtime.Object {
-	return &iamv0.IdentityDisplayResults{}
+	return iamv0.DisplayResource.NewFunc()
```
Comment (ryantxu) [inline pkg/registry/apis/iam/user/rest_display.go:38]:
```suggestion
	return &iamv0.DisplayList{}
```

File: pkg/registry/apis/iam/user/rest_display.go [hunk 3]
```diff
@@ -45,20 +45,19 @@ func (r *LegacyDisplayREST) NamespaceScoped() bool {
 }
 
 func (r *LegacyDisplayREST) GetSingularName() string {
-	// not actually used anywhere, but required by SingularNameProvider
-	return "identitydisplay"
+	return iamv0.DisplayResource.GetSingularName()
```
Comment (ryantxu) [inline pkg/registry/apis/iam/user/rest_display.go:48]: pretty sure this is never used so anything is fine

**Review by ryantxu — APPROVED** [review]

**Pull request merged**